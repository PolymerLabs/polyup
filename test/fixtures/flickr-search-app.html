<!DOCTYPE html>
<html>
<head>
  <title>A simple flickr search example</title>
  <script src='components/webcomponentsjs/webcomponents.js'></script>
  <link rel='import' href='components/polymer/polymer.html'>
  <link rel='import' href='components/core-ajax/core-ajax.html'>
  <link rel='import' href='components/paper-spinner/paper-spinner.html'>
  <style>
    body {
      margin: 40px;
    }
  </style>
</head>
<body>


<polymer-element name='my-debouncer' attributes='input output delay'>
  <script>
    Polymer({
      delay: 300,
      ready: function() {
        this.setOutput();
      },
      inputChanged: function() {
        this.job('inputChanged', this.setOutput, this.delay);
      },
      setOutput: function() {
        this.output = this.input;
      }
    })
  </script>
</polymer-element>

<polymer-element name='flickr-search-app'>
  <template>
    <style>
      paper-spinner {
        display: block;
      }
      img {
        display: inline-block;
        border: 5px solid black;
      }
      img.america {
        border-color: blue;
      }
      img.britain {
        border-color: red;
      }
      img.unknown {
        border-color: gray;
      }
      label {
        display: block;
        padding-bottom: 6px;
      }
      input {
        width: 100%;
        max-width: 500px;
        font-size: 16px;
        padding: 8px;
      }
    </style>

    <label>Search the public domain on Flickr:</label>
    <input autofocus value='{{search}}'>
    <my-debouncer input='{{search}}' output='{{debouncedSearch}}'>
    </my-debouncer>
    <core-ajax
        auto url='https://www.flickr.com/services/rest/' handleAs='json'
        params='{{ {method: "flickr.photos.search", api_key: apiKey, text:debouncedSearch, license:"7,8", format: "json", nojsoncallback: 1 } }}'
        response='{{response}}' error='{{error}}' loading='{{loading}}'>
    </core-ajax>
    <paper-spinner active='{{loading}}'></paper-spinner>
    <template repeat='{{photo in response.photos.photo}}'>
      <img class='{{photo | getClasses | tokenList}}' src='https://farm{{photo.farm}}.staticflickr.com/{{photo.server}}/{{photo.id}}_{{photo.secret}}.jpg'>
    </template>
    <template if='{{error}}'>
      {{error.statusCode}} Error:
      <pre>{{error.response}}</pre>
    </template>
  </template>
  <script>
    Polymer({
      search: 'house',
      apiKey: '5358d9830b6865a13d251e5e1acb4c30',
      getClasses: function(photo) {
        var classes = {};
        if (photo.title.match(/(U\.S\.|USA|America)/i)) {
          classes['america'] = true;
        } else if (photo.title.match(/Britain|U\.K\.|British/i)) {
          classes['britain'] = true;
        } else {
          classes['unknown'] = true;
        }
        return classes;
      }
    });
  </script>
</polymer-element>



<flickr-search-app></flickr-search-app>

</body>
</html>
