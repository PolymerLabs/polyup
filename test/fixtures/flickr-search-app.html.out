<!DOCTYPE html><html><head>
  <title>A simple flickr search example</title>
  <script src="components/webcomponentsjs/webcomponents-lite.js"></script>
  <link rel="import" href="components/polymer/polymer.html">
  <link rel="import" href="components/iron-ajax/iron-ajax.html">
  <link rel="import" href="components/paper-spinner/paper-spinner.html">
  <style>
    body {
      margin: 40px;
    }
  </style>
</head>
<body>


<dom-module id="my-debouncer">
  <script>
    Polymer({
      is: 'my-debouncer',
      properties: {
        delay: {
          type: Number,
          value: 300,
          notify: true
        },
        input: {
          notify: true,
          observer: 'inputChanged'
        },
        output: { notify: true }
      },
      ready: function () {
        this.setOutput();
      },
      inputChanged: function () {
        this.debounce('inputChanged', this.setOutput, this.delay);
      },
      setOutput: function () {
        this.output = this.input;
      }
    });
  </script>
</dom-module>

<dom-module id="flickr-search-app">
  <style>
      paper-spinner {
        display: block;
      }
      img {
        display: inline-block;
        border: 5px solid black;
      }
      img.america {
        border-color: blue;
      }
      img.britain {
        border-color: red;
      }
      img.unknown {
        border-color: gray;
      }
      label {
        display: block;
        padding-bottom: 6px;
      }
      input {
        width: 100%;
        max-width: 500px;
        font-size: 16px;
        padding: 8px;
      }
    </style>
  <template>

    <label>Search the public domain on Flickr:</label>
    <input autofocus="" value="{{search::input}}">
    <my-debouncer input="{{search}}" output="{{debouncedSearch}}">
    </my-debouncer>
    <iron-ajax auto="" url="https://www.flickr.com/services/rest/" handle-as="json" params="{{_computeParams(apiKey, debouncedSearch)}}" last-response="{{response}}" last-error="{{error}}" loading="{{loading}}">
    </iron-ajax>
    <paper-spinner active="{{loading}}"></paper-spinner>
    <template is="dom-repeat" items="{{response.photos.photo}}" as="photo">
      <img src="{{_computeSrc(photo)}}" class$="{{_computeClass(photo)}}">
    </template>
    <template is="dom-if" if="{{error}}">
      <span>{{error.statusCode}}</span> Error:
      <pre>{{error.response}}</pre>
    </template>
  </template>
  <script>
    Polymer({
      is: 'flickr-search-app',
      properties: {
        apiKey: {
          type: String,
          value: '5358d9830b6865a13d251e5e1acb4c30'
        },
        search: {
          type: String,
          value: 'house'
        }
      },
      getClasses: function (photo) {
        var classes = {};
        if (photo.title.match(/(U\.S\.|USA|America)/i)) {
          classes['america'] = true;
        } else if (photo.title.match(/Britain|U\.K\.|British/i)) {
          classes['britain'] = true;
        } else {
          classes['unknown'] = true;
        }
        return classes;
      },
      _computeParams: function (apiKey, debouncedSearch) {
        return {
          method: 'flickr.photos.search',
          api_key: apiKey,
          text: debouncedSearch,
          license: '7,8',
          format: 'json',
          nojsoncallback: 1
        };
      },
      _computeClass: function (photo) {
        return this.tokenList(this.getClasses(photo));
      },
      _computeSrc: function (photo) {
        return 'https://farm' + photo.farm + '.staticflickr.com/' + photo.server + '/' + photo.id + '_' + photo.secret + '.jpg';
      },
      tokenList: function (obj) {
        var pieces = [];
        for (var key in obj) {
          if (obj[key]) {
            pieces.push(key);
          }
        }
        return pieces.join(' ');
      }
    });
  </script>
</dom-module>



<flickr-search-app></flickr-search-app>



</body></html>
